use simple definition of ARRAY_SIZE macro when using Intel compilers
author: Kenneth Hoste (HPC-UGent)
--- blender-2.77a/source/blender/blenlib/BLI_utildefines.h.orig	2016-09-07 14:21:00.287130174 +0200
+++ blender-2.77a/source/blender/blenlib/BLI_utildefines.h	2016-09-07 14:21:08.087031653 +0200
@@ -435,7 +435,7 @@
 	} (void)0
 
 /* assuming a static array */
-#if defined(__GNUC__) && !defined(__cplusplus) && !defined(__clang__)
+#if defined(__GNUC__) && !defined(__cplusplus) && !defined(__clang__) && !defined(__INTEL_COMPILER)
 #  define ARRAY_SIZE(arr) \
 	((sizeof(struct {int isnt_array : ((const void *)&(arr) == &(arr)[0]);}) * 0) + \
 	 (sizeof(arr) / sizeof(*(arr))))
